bin/shellmock

```diff
@@ -38,16 +38,55 @@ skipIfNot()
     fi
 }
 
+#----------------------------------------------------------------------------
+# This function creates a single string containing all $*.  In the process
+# it checks to see if an arg contains a space and if so then it places
+# double quotes around the argument.
+#
+# Note: Even if single quotes were originally used the string and
+# any matching in the ${capture{@]} arrary will have double quotes around
+# the arguments.
+#----------------------------------------------------------------------------
+shellmock_normalize_args() {
+    shellmock_debug "shellmock_normalize_args: before *$**"
+
+    local re="[[:space:]]+"
+    args=""
+    for arg in "${@}";do
+      if [[ $arg =~ $re ]]; then
+          shellmock_debug "shellmock_normalize_args: found space: $arg"
+          args="$args \"$arg\""
+          shellmock_debug "shellmock_normalize_args: args: $args"
+      else
+          args="$args $arg"
+      fi
+      shellmock_debug "shellmock_normalize_args: args: $args"
+    done
+    args=`echo $args|$AWK '{$1=$1;print}'`
```

Comment:
Is there a reason why you're using the backtick notation for command substitution rather than `$(..)` ? I've seen both notations throughout the code and was wondering what was the rationale to use one over the other

Reply:
I can certainly see that I was inconsistent.  I typically use the ` because that is what I had learned first I think.  However, I can not explain the inconsistent usage.

I can update the code to prefer $().

Reply:
I've been using [shellcheck](https://github.com/koalaman/shellcheck) to lint my bash code in the last months. Would it be something worth adding ?

Reply:
I will look into that.